app/scripts/modules/amazon/pipeline/stages/cloneServerGroup/awsCloneServerGroupStage.js

```diff
@@ -93,5 +95,30 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.aws.cloneServerGr
     this.processIsSuspended = (process) => {
       return stage.suspendedProcesses && stage.suspendedProcesses.includes(process);
     };
+
+    this.blockDeviceMappingsSource = () => {
```

Comment:
Generally, in Angular, for something that's going to be called every time a user interacts with the page, you want to manage the state of the field elsewhere, like the `viewState` object.

That would add to the complexity here (you'd have to initialize it, then track changes), and this page (pipeline config) is relatively lightweight, so I'm good with the code as-is.